system-addon/test/unit/lib/Screenshots.test.js

```diff
@@ -102,4 +119,15 @@ describe("Screenshots", () => {
       assert.propertyVal(str, 999999, "\u0009");
     });
   });
+
+  describe("#_shouldGetScreenshots", () => {
+    it("should use private browsing utils to determine if a window is private", () => {
+      Screenshots._shouldGetScreenshots();
+      assert.calledOnce(global.PrivateBrowsingUtils.isWindowPrivate);
+    });
+    it("should return true if there exists at least 1 non-private window", () => {
+      global.PrivateBrowsingUtils.isWindowPrivate = () => false;
+      assert.isTrue(Screenshots._shouldGetScreenshots());
+    });
```

Comment:
A test for the `false` case would be good

Reply:
Still figuring out how to break out of that while(true) loop if since it'll go on forever in this test due to ```.hasMoreElements()``` being mocked to be forever true 😞

Reply:
Could stub `getEnumerator` to return something different for a single test?
```js
let more = 2;
sandbox.stub(Services.wm, "getEnumerator").returns({hasMoreElements() { return more--; }});
```

Set more based on testing 2 private windows, 1, 0, etc.